flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamMockEnvironment.java

```diff
@@ -215,6 +189,31 @@ public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
 		}
 	}
 
+	private <T> void addBufferToOutputList(RecordDeserializer<DeserializationDelegate<T>> recordDeserializer,
```

Comment:
Not that it was commented before, but you might want to add a high level comment about what happens here.